tools/cmake/mbed_greentea.cmake

```diff
@@ -70,6 +81,37 @@ macro(mbed_greentea_add_test)
 
     mbed_set_post_build(${MBED_GREENTEA_TEST_NAME})
 
+    if(NOT ${MBED_OUTPUT_EXT} STREQUAL "")
+        set(MBED_GREENTEA_TEST_IMAGE_NAME "${MBED_GREENTEA_TEST_NAME}.${MBED_OUTPUT_EXT}")
+    else()
+        set(MBED_GREENTEA_TEST_IMAGE_NAME "${MBED_GREENTEA_TEST_NAME}.bin")
+    endif()
+
+    if(DEFINED MBED_GREENTEA_HOST_TESTS_DIR)
+        list(APPEND MBED_HTRUN_OPTIONAL_ARGUMENTS "-e;${MBED_GREENTEA_HOST_TESTS_DIR}")
+    endif()
+
+    if(DEFINED MBED_TARGET)
+        list(APPEND MBED_HTRUN_OPTIONAL_ARGUMENTS "-m;${MBED_TARGET}")
```

Comment:
I think
```cmake
list(APPEND MBED_HTRUN_OPTIONAL_ARGUMENTS -m ${MBED_TARGET})
```
has the same effect as lists are semicolon separated. And we may always assume MBED_TARGET to be defined as greentea tests run on device and require mbed_config.cmake.

Reply:
It does have the same effect, so it's a style point. I'm fine with the way it is now.

I also don't like assuming MBED_TARGET is always defined, as I think we want to break any hard dependencies on config going forward. We assume MBED_CONFIG_DEFINITIONS for extracting the baud rate, but we also check if we've actually found a valid value before passing it to htrun. So I'd prefer to go with the more defensive approach here.

Reply:
This makes sense